build/android/BUILD.gn

```diff
@@ -0,0 +1,340 @@
+  script = "//xwalk/build/android/gn/generate_xwalk_core_library.py"
+  args = [
+    "-s",
+    "$src_dir",
+    "-t",
+    "./",
+  ]
+  outputs = [
+    "$root_out_dir/xwalk_core_library",
+  ]
+  deps = [
+    ":xwalk_core_library_java",
+    "//xwalk/runtime/android/core_shell:xwalk_core_shell_apk",
+  ]
+}
+
+action("xwalk_shared_library") {
+  script = "//xwalk/build/android/gn/generate_xwalk_core_library.py"
+  args = [
+    "-s",
+    "$src_dir",
+    "-t",
+    "./",
+    "--shared",
+  ]
+  outputs = [
+    "$root_out_dir/xwalk_shared_library",
+  ]
+  deps = [
+    ":xwalk_core_library_java",
+  ]
+}
+
+action("xwalk_core_library_java") {
+  script = "//xwalk/build/android/gn/merge_xwalk_core_library_jars.py"
+  args = [
+    "--jar_type=core_library",
+    "--output_dir=$build_dir",
+  ]
+  outputs = [
+    "$root_out_dir/lib.java/xwalk_core_library_java.jar",
+  ]
+  deps = [
+    ":xwalk_core_library_java_app_part",
+    ":xwalk_core_library_java_library_part",
+  ]
+}
+
+action("xwalk_core_library_java_library_part") {
+  script = "//xwalk/build/android/gn/merge_xwalk_core_library_jars.py"
+  args = [
+    "--jar_type=library_part",
+    "--output_dir=$build_dir",
+  ]
+  outputs = [
+    "$root_out_dir/lib.java/xwalk_core_library_java_library_part.jar",
+  ]
+  deps = [
+    "//third_party/android_protobuf:protobuf_nano_javalib",
+    "//third_party/cardboard-java:cardboard-java",
+    "//xwalk/runtime/android/core_internal:xwalk_core_internal_java",
+    "//xwalk/runtime/android/core_internal_empty:xwalk_core_internal_empty_embedder_apk",
+  ]
+}
+
+action("xwalk_core_library_java_app_part") {
+  script = "//xwalk/build/android/gn/merge_xwalk_core_library_jars.py"
+  args = [
+    "--jar_type=app_part",
+    "--output_dir=$build_dir",
+  ]
+  outputs = [
+    "$root_out_dir/lib.java/xwalk_core_library_java_app_part.jar",
+  ]
+  deps = [
+    "//xwalk/runtime/android/core:xwalk_core_java",
+  ]
+}
+
+action("xwalk_core_library_aar") {
+  script = "//xwalk/build/android/gn/generate_xwalk_core_library_aar.py"
+  args = [
+    "-t",
+    "./",
+  ]
+  outputs = [
+    "$root_out_dir/xwalk_core_library.aar",
+  ]
+  deps = [
+    ":xwalk_core_library",
+    ":xwalk_core_library_pom_gen",
+    "//xwalk/runtime/android/core_internal_empty:xwalk_core_empty_embedder_apk",
+  ]
+}
+
+action("xwalk_shared_library_aar") {
+  script = "//xwalk/build/android/gn/generate_xwalk_core_library_aar.py"
+  args = [
+    "-t",
+    "./",
+    "--shared",
+  ]
+  outputs = [
+    "$root_out_dir/xwalk_shared_library.aar",
+  ]
+  deps = [
+    ":xwalk_shared_library",
+    ":xwalk_shared_library_pom_gen",
+    "//xwalk/runtime/android/core_internal_empty:xwalk_core_empty_embedder_apk",
+  ]
+}
+
+# Remove "build/android/maven_pom.gypi" and call "build/util/version.py" directly
+action("xwalk_core_library_pom_gen") {
+  script = "//build/util/version.py"
+  args = [
+    "-i",
+    "$src_dir/xwalk/runtime/android/maven/xwalk_core_library.pom.xml.in",
```

Comment:
You're not declaring this as an input to the action, so if it changes the action will not be run. In fact, you should try using the `process_version` template in chrome/version.gni instead.